code/modules/experisci/experiment/handlers/experiment_handler.dm

```diff
@@ -0,0 +1,393 @@
+	for(var/experiment in GLOB.experiment_handlers)
+		var/datum/component/experiment_handler/experi_handler = experiment
+		var/atom/movable/experi_parent = experi_handler.parent
+		experi_parent.say(message)
+
+/**
+ * Announces a message to this experiment handler
+ *
+ * Arguments:
+ * * message - The message to announce
+ */
+/datum/component/experiment_handler/proc/announce_message(message)
+	var/atom/movable/experi_parent = parent
+	experi_parent.say(message)
+
+/**
+ * Attempts to perform the selected experiment given some arguments
+ */
+/datum/component/experiment_handler/proc/action_experiment(datum/source, ...)
+	// Check if an experiment is selected
+	if (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE))
+		return FALSE
+
+	// Attempt to run
+	var/list/arguments = list(src)
+	arguments = args.len > 1 ? arguments + args.Copy(2) : arguments
+	if (config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE)
+		for (var/datum/experiment/e in linked_web.available_experiments)
+			. = e.actionable(arglist(arguments)) && e.perform_experiment(arglist(arguments))
+	else
+		// Returns true if the experiment was succesfuly handled
+		. = selected_experiment.actionable(arglist(arguments)) && selected_experiment.perform_experiment(arglist(arguments))
+
+/**
+ * Hook for handling UI interaction via signals
+ */
+/datum/component/experiment_handler/proc/ui_handle_experiment(datum/source, mob/user, action)
+	SIGNAL_HANDLER
+	switch(action)
+		if("open_experiments")
+			INVOKE_ASYNC(src, .proc/configure_experiment, null, usr)
+
+/**
+ * Attempts to show the user the experiment configuration panel
+ *
+ * Arguments:
+ * * user - The user to show the experiment configuration panel to
+ */
+/datum/component/experiment_handler/proc/configure_experiment(datum/source, mob/user)
+	ui_interact(user)
+
+/**
+ * Attempts to show the user the experiment configuration panel
+ *
+ * Arguments:
+ * * user - The user to show the experiment configuration panel to
+ */
+/datum/component/experiment_handler/proc/configure_experiment_click(datum/source, mob/user)
+	ui_interact(user)
+
+/**
+ * Attempts to link this experiment_handler to a provided techweb
+ *
+ * This proc attempts to link the handler to a provided techweb, overriding the existing techweb if relevant
+ *
+ * Arguments:
+ * * new_web - The new techweb to link to
+ */
+/datum/component/experiment_handler/proc/link_techweb(datum/techweb/new_web)
+	if (new_web == linked_web)
+		return
+	selected_experiment = null
+	linked_web = new_web
+
+/**
+ * Unlinks this handler from the selected techweb
+ */
+/datum/component/experiment_handler/proc/unlink_techweb()
+	selected_experiment = null
+	linked_web = null
+
+/**
+ * Attempts to link this experiment_handler to a provided experiment
+ *
+ * Arguments:
+ * * e - The experiment to attempt to link to
+ */
+/datum/component/experiment_handler/proc/link_experiment(datum/experiment/experi)
```

Comment:
This here and everywhere else should be written as `experiment`, not `e` or `experi`. Optimize for readability, not writability.